### PR TITLE
Workflow changes to facilitate maintenance

### PR DESCRIPTION
**Pull request by bettinaheim**

### CODE REVIEW

File: .github/workflows/clean_up.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ jobs:
               /repos/nvidia/cuda-quantum/git/refs/heads/$branch 
           done
         env:
-          GH_TOKEN: ${{ github.token }}
+          GH_TOKEN: ${{ secrets.REPO_BOT_ACCESS_TOKEN }}
```
Comment (bmhowe23) [inline .github/workflows/clean_up.yml:45]: Is this new secret pointing back to the public repo? If so, what (if anything) would delete the bot/* branches in the private repo?

Comment (bettinaheim) [reply .github/workflows/clean_up.yml:45]: this is a secret on the repository that the workflow runs on. I can make this more fork-friendly by falling back to the gh token if it is not defined.

**Review by bmhowe23 — APPROVED** [review]
LGTM

**Review by bmhowe23 — APPROVED** [review]
This morning's updates LGTM. I made one comment offline, but I am ok with it as is.

**Pull request merged**